pypeit/scienceimage.py

```diff
@@ -552,7 +536,48 @@ def _get_goodslits(self, maskslits):
 
     # JFH TODO I think science image should not be a child of Processimages. Then this could be running on a file list
     # I think this would be simpler. Implement this!
-    def proc(self, bias_subtract, pixel_flat, bpm, illum_flat=None, apply_gain=True, trim=True,show=False):
+
+    def read_stack(self, files, bias, pixel_flat, bpm, illum_flat, cosmics=False):
+        """  Utility function for reading in image stacks using ProcessImages
+        Parameters
+            file_list:
+            bias:
+            pixel_flat:
+            bpm:
+            illum_flat:
+        Returns:
+        """
+        nfiles = len(files)
+        for ifile in range(nfiles):
```

Comment:
I'm torn on whether to do these operations here or in 
`ProcessImages`.  Depends on whether we'd be doing
similar operations for non-science frames.  Do we know?

Reply:
Let's try to move these methods over in another PR. In general I agree with you.